pennylane/gradients/vjp.py

```diff
@@ -15,11 +15,186 @@
+        # No trainable parameters
+        if jac.shape == (0,):
+            res = qml.math.zeros((1, 0))
+            return res
+        # Single measurement with no dimension e.g. expval
+        if dy.shape == ():
+            jac = math.reshape(qml.math.squeeze(jac), (1, 1))
+            res = math.tensordot(jac, dy_row, [[0], [0]])
+        # Single measurement with dimension e.g. probs
+        else:
+            jac = math.reshape(jac, (len(jac), 1))
+            res = math.tensordot(jac, dy_row, [[0], [0]])
+    # Single measurement with multiple params
+    else:
+        # No trainable parameters (adjoint)
+        if len(jac) == 0:
+            res = qml.math.zeros((1, 0))
+            return res
+        # Single measurement with no dimension e.g. expval
+        if dy.shape == ():
+            jac = qml.math.reshape(qml.math.stack(jac), (1, len(jac)))
+            res = qml.math.tensordot(jac, dy_row, [[0], [0]])
+
+        # Single measurement with dimension e.g. probs
+        else:
+            jac = qml.math.stack(jac)
+            res = qml.math.tensordot(jac, dy_row, [[1], [0]])
+    return res
+
+
+def compute_vjp_multi(dy, jac):
+    """Convenience function to compute the vector-Jacobian product for a given
+    vector of gradient outputs and a Jacobian for a multiple measurements tape.
+
+    Args:
+        dy (tensor_like): vector of gradient outputs
+        jac (tensor_like, tuple): Jacobian matrix
+
+    Returns:
+        tensor_like: the vector-Jacobian product
+
+    **Examples**
+
+    1. For a single parameter and multiple measurement (one without shape and one with shape):
+
+    .. code-block:: pycon
+
+        >>> jac = tuple([np.array(0.1), np.array([0.3, 0.4])])
+        >>> dy = tuple([np.array(1.0), np.array([1.0, 2.0])])
+        >>> compute_vjp_multi(dy, jac)
+        np.array([1.2])
+
+    2. For multiple parameters (in this case 2 parameters) and multiple measurement (one without shape and one with shape):
+
+    .. code-block:: pycon
+
+        >>> jac = tuple([tuple([np.array(0.1), np.array(0.2)]), tuple([np.array([0.3, 0.4]), np.array([0.5, 0.6])])])
+        >>> dy = tuple([np.array(1.0), np.array([1.0, 2.0])])
+        >>> compute_vjp_multi(dy, jac)
+        np.array([1.2, 1.9])
+
+    """
+    if jac is None:
+        return None
+    # Single parameter
+    if not isinstance(jac[0], (tuple, autograd.builtins.SequenceBox)):
+        res = []
+        for i, elem in enumerate(dy):
+            res.append(compute_vjp_single(elem, jac[i]))
+        res = qml.math.stack(res)
+        res = qml.math.stack([math.sum(res)])
+    # Multiple parameters
+    else:
+        res = []
+        for i, elem in enumerate(dy):
+            sub_res = []
+            for j in jac[i]:
+                sub_res.append(qml.math.squeeze(compute_vjp_single(elem, j)))
+            res.append(sub_res)
+        res = qml.math.stack([qml.math.sum(x) for x in zip(*res)])

```

Comment:
I've tried to use `sum` with `axis=0` but unfortunately it does not work with Autograd, because it creates a numpy array of ArrayBoxes and when you stack that it stays a numpy array and become non diff. On the other hand when you have a list of ArrayBoxes, when you stack the list it becomes an ArrayBoxes itself which is differentiable.